### PR TITLE
Test/filter snips

### PR DESCRIPTION
**Pull request by cathmer**
Write tests for filter snips class. Uses the annotation to access private methods, so these methods could be unit tested.

### CODE REVIEW

**Comment by wouterjsmit** [comment]
The tests look good and comments are clear.

Could you squash the `Fix import order` 508f529 and `Put && on newline` 33c2fcc? These do not need to be separate commits. (squash them into the commit where you originally changed that line)

Could you also mention in the commit message of `Set sequence length for nodes` 3f6d191 and `Delete Graph class` bf8f76c why you did what you did? I'm not sure why the Graph class could be deleted or what the stray sequence length did.

I'm sure I'll hear if you you're having Git trouble :stuck_out_tongue:

**Comment by FarisElghlan** [comment]
Next time try to build the project locally before commiting so you have fewer failing builds in a row. The tests look nice.

File: PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java [hunk 1]
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
...
-    // filteredGraph.print();
+    
+    assertEquals(filteredGraph.getSize(), 3);
+    Node root = filteredGraph.getRoot();
+    assertEquals(root.getId(), 1);
+    assertTrue(root.getOutlinks().contains(8));
+    assertTrue(root.getOutlinks().contains(9));
+
+    assertEquals(filteredGraph.getNode(8), graph.getNode(8));
+    assertEquals(filteredGraph.getNode(9), graph.getNode(9));
```
Comment (FarisElghlan) [inline PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java]: Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getNode` separately).

File: PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java [hunk 2]
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
...
+   * 1 -
+   *    - 9
+   */
+  @Test
+  public void testFilter() {
     FilterSnips filter = new FilterSnips(graph);
     OriginalGraph filteredGraph = filter.filter();
-    // filteredGraph.print();
+    
+    assertEquals(filteredGraph.getSize(), 3);
```
Comment (FarisElghlan) [inline PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java]: The first parameter of `assertEquals` should be the expected value and the second parameter should be the actual value:
`assertEquals(3, filteredGraph.getSize());`

This comment also applies to a lot of your other assert statements

File: PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java [hunk 3]
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
...
+        filter, graph.getNode(1));
+
+    assertEquals(snip.getOutlinks().size(), 2);
+    assertTrue(snip.getOutlinks().contains(8));
+    assertTrue(snip.getOutlinks().contains(9));
+    
+    assertEquals(snip.getInlinks().size(), 0);
+    
+    assertEquals(snip.getId(), 1);
+    assertEquals(snip.getSnips(), 2);
```
Comment (FarisElghlan) [inline PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java]: Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getInlinks` and `getId/getSnips` separately).

File: PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java [hunk 4]
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
...
+    assertEquals(snip.getOutlinks().size(), 2);
+    assertTrue(snip.getOutlinks().contains(8));
+    assertTrue(snip.getOutlinks().contains(9));
+    
+    assertEquals(snip.getInlinks().size(), 2);
+    assertTrue(snip.getInlinks().contains(2));
+    assertTrue(snip.getInlinks().contains(3));
+    
+    assertEquals(snip.getId(), 4);
+    assertEquals(snip.getSnips(), 1);
```
Comment (FarisElghlan) [inline PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java]: Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getInlinks` and `getId / getSnips` separately).

**Comment by cathmer** [comment]
@Pathemeous @FarisElghlan I squashed some commits together to make it look cleaner. I've also fixed the assertEquals, so that the first input is the expected value and the second input is the real value.

I didn't make more test methods to split up the asserts of a method, because I feel this is redundant and a waste of time, because the tests do check what they have to check, and it works. 

Also, the last commit has no value, but was used to check if it would make Travis build. It turned out Travis just needed some time (like an hour) to start building..

**Comment by wouterjsmit** [comment]
@cathmer That sometimes happens, because Travis is free and their resources aren't endless. Very weird though.

Lastly, could I ask you to remove the last commit, since I really do not want that in the history.
Can be done with rebasing (remove the whole line for that commit)
or by resetting the tip of the branch one commit back:
`git reset --hard HEAD~1`

Followed by a `git push -f` to update the remote.

Git note:
One last thing that I'm also just noticing now. When you update dev into your local branch with a merge, you get an empty merge commit (because there were no conflicts and it could be fast-forwarded).
Locally, if you instead rebase your local branch onto `dev` (taking the tip of `dev` and addding your commits to there, rather than where you branched of before), this commit disappears (even after you have already done the merge.

I would ask you to rebase to remove the empty merge commit from `dev`, but I'm interested to see if github automatically removes that commit when this PR is merged.

**Comment by cathmer** [comment]
The empty commit has been removed. I think it's ready for merging now.

**Comment by wouterjsmit** [comment]
@cathmer Not sure what happened, but now I see some commits from `dev` on top of your changes.

Edit: If I check out your branch and perform the following two commands, I get the desired result, where only your changes are put on top of `dev`.
`git fetch origin && git rebase origin/dev`

**Comment by cathmer** [comment]
Done, now the other commits are gone.

**Pull request merged**